src/modules/auth/auth.ts

```diff
@@ -163,6 +170,24 @@ export class Auth {
     }
   }
 
+  async initiateClientFlow(tokenGetParam?: any) {
+      if (!this.#onAuthSuccess) {
+        throw new Error(
+          "'onAuthSuccess' was not provided. Please provide the 'onAuthSuccess' function"
+        );
+      }
+      const payload: ClientTokenRequestPayload = {
+        client_id: this.#clientId!,
+        client_secret: this.#clientSecret!,
+        grant_type: GrantType.CLIENT_CREDENTIALS
+      };
+      const result = await this.#post("/token", payload);
+      const { access_token: accessToken, refresh_token: refreshToken } = result;
```

Comment:
Sorry, I missed this the first time. We don't provide refresh token in the client credentials flow. The response will contain only `access_token: str`, `expires_in: int` and `token_type: str`
```suggestion
      const { access_token: accessToken } = result;
```

Let's also update the return type definition of the `this.#post` method, and update the parameter type of `onSuccess` function here to optional https://github.com/newscred/Welcome-Nodejs-SDK/blob/3dc7847a6b4483acfca1e0ed6f4d1bb2b0561612/src/welcome.ts#L19

```ts
refreshToken?: string
```

Reply:
I noticed this in testing, should have confirmed whether that behavior is expected. Making this change.